openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,219 @@ def maintained_selection():
+
+    cmds.refresh(force=True)
+
+    refreshFrameInt = int(cmds.playbackOptions(q=True, minTime=True))
+    cmds.currentTime(refreshFrameInt - 1, edit=True)
+    cmds.currentTime(refreshFrameInt, edit=True)
+
+    # Use displayLights setting from instance
+    key = "displayLights"
+    preset["viewport_options"][key] = instance.data[key]
+
+    # Override transparency if requested.
+    transparency = instance.data.get("transparency", 0)
+    if transparency != 0:
+        preset["viewport2_options"]["transparencyAlgorithm"] = transparency
+
+    # Isolate view is requested by having objects in the set besides a
+    # camera. If there is only 1 member it'll be the camera because we
+    # validate to have 1 camera only.
+    if instance.data["isolate"] and len(instance.data["setMembers"]) > 1:
+        preset["isolate"] = instance.data["setMembers"]
+
+    # Show/Hide image planes on request.
+    image_plane = instance.data.get("imagePlane", True)
+    if "viewport_options" in preset:
+        preset["viewport_options"]["imagePlane"] = image_plane
+    else:
+        preset["viewport_options"] = {"imagePlane": image_plane}
+
+    # Disable Pan/Zoom.
+    pan_zoom = cmds.getAttr("{}.panZoomEnabled".format(preset["camera"]))
+    preset.pop("pan_zoom", None)
+    preset["camera_options"]["panZoomEnabled"] = instance.data["panZoom"]
+
+    # Need to explicitly enable some viewport changes so the viewport is
+    # refreshed ahead of playblasting.
+    keys = [
+        "useDefaultMaterial",
+        "wireframeOnShaded",
+        "xray",
+        "jointXray",
+        "backfaceCulling",
+        "textures"
+    ]
+    viewport_defaults = {}
+    for key in keys:
+        viewport_defaults[key] = cmds.modelEditor(
+            instance.data["panel"], query=True, **{key: True}
+        )
+        if preset["viewport_options"][key]:
+            cmds.modelEditor(
+                instance.data["panel"], edit=True, **{key: True}
+            )
+
+    override_viewport_options = (
+        capture_preset["Viewport Options"]["override_viewport_options"]
+    )
+
+    # Force viewer to False in call to capture because we have our own
+    # viewer opening call to allow a signal to trigger between
+    # playblast and viewer
+    preset["viewer"] = False
+
+    # Update preset with current panel setting
+    # if override_viewport_options is turned off
+    if not override_viewport_options:
+        panel_preset = capture.parse_view(instance.data["panel"])
+        panel_preset.pop("camera")
+        preset.update(panel_preset)
+
+    path = capture_with_preset(
+        preset, instance)
+
+    # Restoring viewport options.
+    if viewport_defaults:
+        cmds.modelEditor(
+            instance.data["panel"], edit=True, **viewport_defaults
+        )
+
+    try:
+        cmds.setAttr(
+            "{}.panZoomEnabled".format(preset["camera"]), pan_zoom)
+    except RuntimeError:
+        self.log.warning("Cannot restore Pan/Zoom settings.")
```

Comment:
Setting the pan zoom state would be better in a context manager and likely closer to where the state must be set (e.g. in the `capture` fucntion itself.)